ios/LayoutReanimation/REASwizzledUIManager.mm

```diff
@@ -16,76 +17,65 @@
 #endif
 
 @interface RCTUIManager (REA)
-- (void)_manageChildren:(NSNumber *)containerTag
-        moveFromIndices:(NSArray<NSNumber *> *)moveFromIndices
-          moveToIndices:(NSArray<NSNumber *> *)moveToIndices
-      addChildReactTags:(NSArray<NSNumber *> *)addChildReactTags
-           addAtIndices:(NSArray<NSNumber *> *)addAtIndices
-        removeAtIndices:(NSArray<NSNumber *> *)removeAtIndices
-               registry:(NSMutableDictionary<NSNumber *, id<RCTComponent>> *)registry;
-
-- (RCTViewManagerUIBlock)uiBlockWithLayoutUpdateForRootView:(RCTRootShadowView *)rootShadowView;
-
 - (NSArray<id<RCTComponent>> *)_childrenToRemoveFromContainer:(id<RCTComponent>)container
                                                     atIndices:(NSArray<NSNumber *> *)atIndices;
 @end
 
-@implementation REAUIManager {
-  NSMutableDictionary<NSNumber *, NSMutableSet<id<RCTComponent>> *> *_toBeRemovedRegister;
-  NSMutableDictionary<NSNumber *, NSNumber *> *_parentMapper;
+@implementation REASwizzledUIManager {
   REAAnimationsManager *_animationsManager;
-  std::weak_ptr<reanimated::Scheduler> _scheduler;
+  RCTUIManager *_uiManager;
 }
 
-+ (NSString *)moduleName
-{
-  return NSStringFromClass([RCTUIManager class]);
-}
+static REASwizzledUIManager *_reaUIManager;
 
-- (void)invalidate
+- (instancetype)initWithUIManager:(RCTUIManager *)uiManager
+            withAnimatioinManager:(REAAnimationsManager *)animationsManager
 {
-  [_animationsManager invalidate];
-  [super invalidate];
+  if (self = [super init]) {
+    _animationsManager = animationsManager;
+    _uiManager = uiManager;
+    _reaUIManager = self;
```

Comment:
I don't quite understand the need for this whole REASwizzledUIManager class to exists. Also, assigning singletons like that seem a bit risky.

It seem to me that the reason it's been done this way was for the swizzled methods to be able to access _uiManager and _animationsManager fields. I think the former can just be replaced by `self` as the swizzling concept refers to swapping out implementation of certain methods. The UI manager object will be the same and the swizzled method self will refer to it.

As for the latter field, the best way IMO would be to use category on UIManager class that adds animationsManager property to it using associated object technique (https://nshipster.com/associated-objects/)

Reply:
1. I've made the changes based on your suggestions 👍. I dumped the singleton and used associated objects with the category instead.
2. The only reason I created the `REASwizzledUIManager` class was to keep the swizzled methods separate from the animation manager. I didn't want to put everything into one file. But yeah, if you think it's best, I can move the `_manageChildren` and `_uiBlockWithLayoutUpdateForRootView `methods back to the `REAAnimationManager` class.